riak_test/tests/regression_tests.erl

```diff
@@ -145,6 +147,85 @@ verify_cs512(UserConfig, BucketName) ->
     assert_notfound(UserConfig,BucketName),
     ok.
 
+verify_cs770({UserConfig, {RiakNodes, _, _}}, BucketName) ->
+    %% put object and cancel it;
+    ?assertEqual(ok, erlcloud_s3:create_bucket(BucketName, UserConfig)),
+    Key = "foobar",
+    lager:debug("starting cs770 verification: ~s ~s", [BucketName, Key]),    
+
+    {ok, Socket} = rtcs_object:upload(UserConfig,
+                                      {normal_partial, 3*1024*1024, 1024*1024},
+                                      BucketName, Key),
+    
+    [[{UUID, M}]] = get_manifests(RiakNodes, BucketName, Key),
+
+    %% Even if CS is smart enough to remove canceled upload, at this
+    %% time the socket will be still alive, so no cancellation logic
+    %% shouldn't be triggerred.
+    ?assertEqual(writing, M?MANIFEST.state),
+    lager:debug("UUID of ~s ~s: ~p", [BucketName, Key, UUID]),
+
+    %% Emulate socket error with {error, closed} at server
+    ok = gen_tcp:close(Socket),
+    %% This wait is just for convenience
+    timer:sleep(1000),
+    retry(8, 4096,
+          fun() ->
+                  [[{UUID, Mx}]] = get_manifests(RiakNodes, BucketName, Key),
+                  scheduled_delete =:= Mx?MANIFEST.state
+          end),
+
+    Pbc = rt:pbc('dev1@127.0.0.1'),
+
+    %% verify that object is also stored in latest GC bucket
+    Ms = all_manifests_in_gc_bucket(Pbc),
+    lager:info("Retrieved ~p manifets from GC bucket", [length(Ms)]),
+    ?assertMatch(
+       [{UUID, _}],
+       lists:filter(fun({UUID0, M1}) when UUID0 =:= UUID ->
+                            ?assertEqual(pending_delete, M1?MANIFEST.state),
+                            true;
+                       ({UUID0, _}) ->
+                            lager:debug("UUID=~p / ~p",
+                                        [mochihex:to_hex(UUID0), mochihex:to_hex(UUID)]),
+                            false;
+                       (_Other) ->
+                            lager:error("Unexpected: ~p", [_Other]),
+                            false
+                    end, Ms)),
+
+    lager:info("cs770 verification ok", []),
+    ?assertEqual(ok, erlcloud_s3:delete_bucket(BucketName, UserConfig)),
+    ok.
+
+retry(0, _, _) ->
```

Comment:
No continuation needed like `rtcs:wait_until/4`, so just `rt:wait_until/3` might be cleaner.